### PR TITLE
Create drawio in ibmvpc

### PR DESCRIPTION
**Pull request by haim-kermany**
1. GenerateDrawioTreeNode() for exery resource
2. DrawioGenerator - a new struct in vpcmodel pkg. that generate the treenodes
3. rewrite DrawioOutputFormatter - now it is much simpler

please notice that DrawioOutputFormatter will be changed according to the grouping information

### CODE REVIEW

File: pkg/ibmvpc/ibmDrawioGenerator.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	}
+	return nil
+}
+
+func (ni *NetworkInterface) GenerateDrawioTreeNode(gen *vpcmodel.DrawioGenerator) drawio.TreeNodeInterface {
+	return drawio.NewNITreeNode(
+		gen.TreeNode(ni.subnet).(drawio.SquareTreeNodeInterface),
+		nil, ni.Name())
+}
+func (n *IKSNode) GenerateDrawioTreeNode(gen *vpcmodel.DrawioGenerator) drawio.TreeNodeInterface {
```
Comment (adisos) [inline pkg/ibmvpc/ibmDrawioGenerator.go:48]:
```suggestion
}

func (n *IKSNode) GenerateDrawioTreeNode(gen *vpcmodel.DrawioGenerator) drawio.TreeNodeInterface {
```

File: pkg/ibmvpc/ibmDrawioGenerator.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	return nil
```
Comment (adisos) [inline pkg/ibmvpc/ibmDrawioGenerator.go:49]: Is this planned to be supported? add some comment?

File: pkg/vpcmodel/drawioGenerator.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+// DrawioResourceIntf is the interface of all the resources that are converted to a drawio treeNodes
+type DrawioResourceIntf interface {
+	GenerateDrawioTreeNode(gen *DrawioGenerator) drawio.TreeNodeInterface
+}
+
+// DrawioGenerator is the struct that generate the drawio tree.
+// its main interface is:
+// 1. TreeNode() - generate and returns the drawio tree node of a resource
+// 2. the constructor - generate the treeNodes that does not represent a specific resource
+// (the constructor creates the publicNetwork tree node, and the Cloud TreeNode)
```
Comment (adisos) [inline pkg/vpcmodel/drawioGenerator.go:16]: explain also what do `Network()` and `PublicNetwork()` return

**Review by adisos — COMMENTED** [review]
looks good! 
Added a few comments

File: pkg/vpcmodel/abstractVPC.go [hunk 4]
```diff
@@ -14,6 +14,7 @@ type VPCResourceIntf interface {
 	// TODO: remove Details and DetailsMap from this interface
 	Details() []string
 	DetailsMap() []map[string]string
+	DrawioResourceIntf
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/vpcmodel/abstractVPC.go:17]: there will be structs that implements DrawioResourceIntf but not VPCResourceIntf.

File: pkg/ibmvpc/ibmDrawioGenerator.go (hunk 1 above)
Comment (haim-kermany) [reply pkg/ibmvpc/ibmDrawioGenerator.go]: fixed

File: pkg/ibmvpc/ibmDrawioGenerator.go (hunk 2 above)
Comment (haim-kermany) [reply pkg/ibmvpc/ibmDrawioGenerator.go:51]: actually, I do not know what it is. 
adding comment

**Review by adisos — APPROVED** [review]

**Pull request merged**